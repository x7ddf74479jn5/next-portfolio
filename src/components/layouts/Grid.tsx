import styles from "src/styles/components/layouts/Grid.module.scss";

export const GridRow: React.FC = ({ children }) => {
  return <div className={styles.gridRow}>{children}</div>;
};
