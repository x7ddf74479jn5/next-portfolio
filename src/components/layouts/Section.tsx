import clsx from "clsx";
import styles from "src/styles/components/layouts/Section.module.scss";

type Props = {
  children: React.ReactNode;
  background?: "primary" | "secondary" | "light";
  id?: string;
  wide?: boolean;
};

export const WrapInSection: React.FC<Props> = ({ children, background, wide = false, ...rest }) => {
  const outerClassName = clsx(styles.section, {
    [styles.bgPrimary]: background === "primary",
    [styles.bgSecondary]: background === "secondary",
    [styles.bgLight]: background === "light",
  });
  const innerClassName = wide ? styles.sectionWrapInWide : styles.sectionWrapIn;
  return (
    <section className={outerClassName} {...rest}>
      <div className={innerClassName}>{children}</div>
    </section>
  );
};
