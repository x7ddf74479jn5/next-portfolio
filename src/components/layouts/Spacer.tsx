import clsx from "clsx";
import styles from "src/styles/components/layouts/Spacer.module.scss";

type Props = {
  size: "xxxs" | "xxs" | "xs" | "sm" | "md" | "lg" | "xl";
};

const Spacer: React.VFC<Props> = ({ size = "md" }) => {
  return (
    <div
      className={clsx({
        [styles.sm]: size === "sm",
        [styles.md]: size === "md",
        [styles.lg]: size === "lg",
      })}
    ></div>
  );
};

export default Spacer;
