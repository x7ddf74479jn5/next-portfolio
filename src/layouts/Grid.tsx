import styles from "src/styles/layouts/Grid.module.scss";

export const GridRow: React.FC = ({ children }) => {
  return <div className={styles.gridRow}>{children}</div>;
};
