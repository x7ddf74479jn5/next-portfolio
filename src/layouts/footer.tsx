// import Image from "next/image";
import Link from "next/link";
import { FooterMenus } from "src/components/common/footer/FooterMenus";
import PandasharkLogo from "src/components/common/PandasharkLogo";
import styles from "src/styles/components/common/PandasharkLogo.module.scss";

export const Footer: React.FC = () => {
  const handle = (event: React.MouseEvent<HTMLHeadingElement, MouseEvent>) => {};

  return (
    <footer className="footer">
      <div className="main c-section">
        <div className="main-inner">
          <section className={styles.logoArea}>
            <PandasharkLogo width={128} height={64} />
            {/* <h1 className={styles.productlogo}>
              <Link href="/">
                <a target="_self">
                  <Image
                    width={128}
                    height={64}
                    alt="パンダシャーク"
                    className="logo-img"
                    src="/img/icons/pandashark_logo_rectangle.png"
                  />
                </a>
              </Link>
            </h1> */}

            {/* <a href="/" target="_self">
                <img alt="パンダシャーク" className="logo-img" src="/img/icons/pandashark_logo.png" />
              </a> */}
            {/* <a className="p-btn-round p-icon__arrow-next" href="/community/" target="_self" role="button">
              Webを学びたい？
            </a> */}
          </section>
          <FooterMenus />

          <section className="foot">
            <ul>
              <Link href="/terms/">
                <a>
                  <li>利用規約</li>
                </a>
              </Link>
              <Link href="/privacy/">
                <a>
                  <li>プライバシーポリシー</li>
                </a>
              </Link>

              {/* <a href="/terms/"><li>利用規約</li></a>
            <a href="/privacy/"><li>プライバシーポリシー</li></a> */}
            </ul>
          </section>
        </div>
      </div>
    </footer>
  );
};
