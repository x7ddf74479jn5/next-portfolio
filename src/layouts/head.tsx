// <!DOCTYPE html>
// <html lang="ja" dir="ltr" locale="ja_ALL" user-region="north-america">
// <head>
//   <meta charset="utf-8">
//   <meta name="viewport" content="initial-scale=1, minimum-scale=1, width=device-width">
//   <title>Pandashark's portfolio</title>
//   <meta name="description" content="Pandasharkのポートフォリオサイトです。">
//   <link href="/css/style.css" rel="stylesheet">
//   <link href="/css/animate.css" rel="stylesheet">
//   <!-- OGP Tag data-->
//   <meta property="og:description" content="Pandasharkのポートフォリオサイトです。">
//   <meta property="og:title" content="Pandashark's portfolio">
//   <meta property="og:image" content="https://pandashark.web.app/assets/img/icons/logo.png">
//   <meta property="og:image:width" content="600">
//   <meta property="og:image:height" content="315">
//   <meta property="og:type" content="website">
//   <meta property="og:url" content="https://pandashark.web.app/">
//   <meta name="twitter:card" content="summary_large_image" />
//   <meta name="twitter:site" content="@pandashark_" />
//   <link rel="shortcut icon" href="/img/icons/favicon.ico">
//   <link rel="apple-touch-icon" sizes="180x180" href="/img/icons/apple-touch-icon.png">
//   <link rel="icon" type="image/png" sizes="32x32" href="/img/icons/favicon-32x32.png">
//   <link rel="icon" type="image/png" sizes="16x16" href="/img/icons/favicon-16x16.png">
//   <link rel="manifest" href="/public/site.webmanifest">
//   <meta name="msapplication-TileColor" content="#da532c">
//   <meta name="theme-color" content="#181a1b">
//   <!-- Carousel slider -->
//   <link href="https://cdnjs.cloudflare.com/ajax/libs/Swiper/4.5.1/css/swiper.min.css" rel="stylesheet">
//   <!-- Font Awesome Icons -->
//   <link href="https://use.fontawesome.com/releases/v5.6.1/css/all.css" rel="stylesheet">
//   <!-- Google Fonts -->
//   <link href="https://fonts.googleapis.com/css?family=Courgette&display=swap&subset=latin-ext" rel="stylesheet">
//   <!-- The core Firebase JS SDK is always required and must be listed first -->
//   <script src="/__/firebase/7.24.0/firebase-app.js"></script>
//   <!-- TODO: Add SDKs for Firebase products that you want to usehttps://firebase.google.com/docs/web/setup#available-libraries -->
//   <script src="/__/firebase/7.24.0/firebase-analytics.js"></script>
//   <script src="/__/firebase/7.24.0/firebase-functions.js"></script>
//   <!-- Initialize Firebase -->
//   <script src="/__/firebase/init.js"></script>
// </head>

import Head from "next/head";

interface Props {
  title: string;
  description: string;
  keyword: string;
  image: string;
  url: string;
  og: {
    type: string;
    image: string;
    imageHight: string;
    imageWidth: string;
    twitterCard: string;
    twitterSite: string;
  };
  icon: string;
  appleTouchIcon: string;
  manifest: string;
}

const headerProps = {
  title: "Pandashark's portfolio",
  description: "Pandasharkのポートフォリオサイトです。",
  keyword: "",
  image: "/img/icons/pandashark_logo.png",
  url: "https://pandashark.web.app/",
  og: {
    type: "website",
    description: "Pandasharkのポートフォリオサイトです。",
    image: "/img/icons/pandashark_logo.png",
    imageHight: "600",
    imageWidth: "315",
    url: "https://pandashark.web.app/",
    twitterCard: "summary_large_image",
    twitterSite: "@pandashark",
  },
  icon: "/img/icons/favicon.ico",
  appleTouchIcon: "/img/icons/apple-touch-icon.png",
  manifest: "",
};

export default ({
  title,
  description,
  keyword,
  image,
  url,
  og,
  icon,
  appleTouchIcon,
  manifest,
}: Props): JSX.Element => {
  return (
    <Head>
      <title>{title}</title>
      <meta property="og:title" content={title} />
      <meta property="og:description" content={description} />
      <meta name="keywords" content={keyword} />
      <meta property="og:type" content="website" />
      <meta property="og:url" content={url} />
      <meta property="og:image" content={image} />
      <meta property="og:image:hight" content={og.imageHight} />
      <meta property="og:image:width" content={og.imageWidth} />
      <meta property="og:site_name" content={title} />
      <meta name="twitter:card" content={og.twitterCard} />
      <meta name="twitter:site" content={og.twitterSite} />
      <meta name="twitter:url" content={url} />
      <meta name="twitter:title" content={title} />
      <meta name="twitter:description" content={description} />
      <meta name="twitter:image" content={image} />
      <meta name="msapplication-TileColor" content="#da532c" />
      <meta name="theme-color" content="#181a1b" />
      <link rel="canonical" href={url} />
      {/* <link rel="stylesheet" href={"https://cdnjs.cloudflare.com/ajax/libs/Swiper/4.5.1/css/swiper.min.css"} /> */}
      <link rel="stylesheet" href={"https://use.fontawesome.com/releases/v5.6.1/css/all.css"} />
      <link rel="stylesheet" href={"https://fonts.googleapis.com/css?family=Courgette&display=swap&subset=latin-ext"} />
      <link rel="shortcut icon" href={icon} />
      <link rel="apple-touch-icon" href={appleTouchIcon} />
      <script
        src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
        integrity="sha256-pasqAKBDmFT4eHoN2ndd6lN370kFiGUFyTiUHWhU7k8="
      ></script>
      <script src="/js/animate.js" type="text/javascript"></script>
    </Head>
  );
};
