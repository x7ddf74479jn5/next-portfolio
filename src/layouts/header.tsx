import * as React from "react";
// import styles from "src/styles/components/common/PandasharkLogo.module.scss";

type HeaderProps = {
  open: boolean;
  isDialogOpen: boolean;
  toggleNav(): void;
  closeNav(e?: React.KeyboardEvent<HTMLDivElement>): void;
  closeDialog(): void;
  openDialog(e?: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void;
};

export const Header: React.VFC<HeaderProps> = ({
  open,
  isDialogOpen,
  toggleNav,
  closeNav,
  closeDialog,
  openDialog,
}) => {
  // const [open, setOpen] = React.useState(false);
  // const classes= {
  //   navOpen:"nav-open"
  // }
  // const toggleNav = () => {
  //   setOpen((prevState) => !prevState);
  // };
  // const closeNav = () => {
  //   setOpen(false);
  // };

  const navOpen = open ? "nav-open" : "";
  return (
    <div className={navOpen}>
      <header className={navOpen + " fixed-nav " + "l-header "}>
        <div className="l-header-inner">
          {/* <div className={styles.productLogo}>
            <PandasharkLogo width={128} height={64} />
          </div> */}

          {/* <!-- Navigation toggle button for Smartphone--> */}
          <button
            className={navOpen + " outside " + "nav-toggle-button "}
            data-about-nav-toggle
            aria-label="ナビゲーション メニューの切り替え"
            id="nav-toggle-button"
            onClick={() => {
              return toggleNav();
            }}
          >
            <img
              alt="ナビゲーション メニューを開く"
              className={navOpen + " menu-bars "}
              src="/img/icons/menu.png"
              height="24"
              width="24"
            />
            <img
              alt="ナビゲーション メニューを閉じる"
              className={navOpen + " menu-close "}
              src="/img/icons/close.png"
              height="24"
              width="24"
            />
          </button>
          {/* <HeaderMenus closeNav={closeNav} closeDialog={closeDialog} openDialog={openDialog} /> */}
        </div>
        <div>
          <div
            className="black-bg"
            role="button"
            id="js-black-bg"
            onKeyDown={(e) => {
              return closeNav(e);
            }}
            onClick={() => {
              return closeNav();
            }}
          ></div>
        </div>
      </header>
    </div>
  );
};
