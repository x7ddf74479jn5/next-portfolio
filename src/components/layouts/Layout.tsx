import * as React from "react";
import { Footer } from "src/components/footer/Footer";
import { Header } from "src/components/header/Header";
import Spacer from "src/components/layouts/Spacer";

export const Layout: React.FC = ({ children }) => {
  return (
    <>
      <Header />
      <Spacer size="md" />
      <main>{children}</main>
      <Footer />
    </>
  );
};
