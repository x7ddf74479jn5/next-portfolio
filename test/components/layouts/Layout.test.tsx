import { Layout } from "src/components/layouts/Layout";
import { render, setupWindow } from "test/test-utils";

describe("Layout", () => {
  beforeAll(() => {
    setupWindow();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Layout>children</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("rendered correctly", () => {
    const { getByRole } = render(<Layout>children</Layout>);
    const targetElement = getByRole("main");
    expect(targetElement).toHaveTextContent("children");
  });
});
