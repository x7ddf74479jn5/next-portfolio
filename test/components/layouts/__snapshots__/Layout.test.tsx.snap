// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="headerInner"
    >
      <div
        class="productLogo"
      >
        <a
          aria-label="パンダシャーク"
          href="/"
          target="_self"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4IiBoZWlnaHQ9IjY0IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <noscript />
            <img
              alt="Pandashark's icon"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </div>
      <button
        aria-label="ナビゲーション メニューの切り替え"
        class="navToggleButton"
        data-about-nav-toggle="true"
        id="nav-toggle-button"
      >
        <svg
          aria-hidden="true"
          aria-label="ナビゲーション メニューを開く\\""
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <nav
        class="headerNavTop"
      >
        <h1
          class="productLogo"
        >
          <a
            aria-label="パンダシャーク"
            href="/"
            target="_self"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="Pandashark's icon"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </a>
        </h1>
        <ul>
          <li>
            <a
              class="linkItem"
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="/samples"
            >
              Samples
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="https://twitter.com/pandashark6"
            >
              <img
                alt="twitter"
                src="/img/icons/twitter-logo.svg"
              />
              <span>
                Twitter
              </span>
            </a>
          </li>
          <li>
            <a
              class="linkItem"
              href="https://github.com/x7ddf74479jn5"
            >
              <img
                alt="github"
                src="/img/icons/github-logo.svg"
              />
              <span>
                GitHub
              </span>
            </a>
          </li>
          <li>
            <button
              class="buttonItem"
            >
              <img
                alt="chatbot"
                src="/img/icons/question_answer_black.svg"
              />
              <span>
                Chat Bot
              </span>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <div
    class="md"
  />
  <main>
    children
  </main>
  <footer
    class="footer"
  >
    <div
      class="main"
    >
      <div
        class="mainInner"
      >
        <div
          class="wrapper"
        >
          <section
            class="logoArea"
          >
            <div
              class="productLogo"
            >
              <a
                aria-label="パンダシャーク"
                href="/"
                target="_self"
              >
                <div
                  style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
                >
                  <div
                    style="box-sizing: border-box; display: block; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      role="presentation"
                      src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4IiBoZWlnaHQ9IjY0IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                      style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                    />
                  </div>
                  <noscript />
                  <img
                    alt="Pandashark's icon"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                </div>
              </a>
            </div>
          </section>
          <section
            class="sitemap"
          >
            <ul
              class="footerLinks"
            >
              <li>
                <details
                  class="accordion"
                >
                  <summary>
                    自己紹介
                  </summary>
                  <ul
                    class="accordionItems"
                  >
                    <li>
                      <a
                        href="/about#vision"
                      >
                        Vision
                      </a>
                    </li>
                    <li>
                      <a
                        href="/about#profile"
                      >
                        Profile
                      </a>
                    </li>
                    <li>
                      <a
                        href="/about#career"
                      >
                        Career
                      </a>
                    </li>
                    <li>
                      <a
                        href="/about#skills"
                      >
                        Skills
                      </a>
                    </li>
                  </ul>
                </details>
              </li>
              <li>
                <details
                  class="accordion"
                >
                  <summary>
                    制作事例
                  </summary>
                  <ul
                    class="accordionItems"
                  >
                    <li>
                      <a
                        href="/samples/portfolio"
                      >
                        ポートフォリオサイト
                      </a>
                    </li>
                    <li>
                      <a
                        href="/samples/ec"
                      >
                        ECサイト
                      </a>
                    </li>
                    <li>
                      <a
                        href="/samples/chat"
                      >
                        Chat App
                      </a>
                    </li>
                  </ul>
                </details>
              </li>
              <li>
                <details
                  class="accordion"
                >
                  <summary>
                    リンク集
                  </summary>
                  <ul
                    class="accordionItems"
                  >
                    <li>
                      <a
                        href="https://twitter.com/pandashark6"
                      >
                        Twitter
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://github.com/x7ddf74479jn5"
                      >
                        Github
                      </a>
                    </li>
                  </ul>
                </details>
              </li>
            </ul>
          </section>
        </div>
        <section
          class="foot"
        >
          <ul>
            <li>
              <a
                href="/terms"
              >
                利用規約
              </a>
            </li>
            <li>
              <a
                href="/privacy"
              >
                プライバシーポリシー
              </a>
            </li>
          </ul>
        </section>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
